python/setup.py

```diff
@@ -174,16 +176,22 @@ def run(self):
       sys.argv.remove(warnings_as_errors)
 
     # C++ implementation extension
-    ext_module_list.append(
+    ext_module_list.extend([
         Extension(
             "google.protobuf.pyext._message",
             glob.glob('google/protobuf/pyext/*.cc'),
             include_dirs=[".", "../src"],
-            libraries=['protobuf'],
+            extra_objects=['../src/.libs/libprotobuf.a',
```

Comment:
I don't think this is safe to do in the general case, because it won't work unless you've specially built protobuf as -fPIC. It seems best for now to hide this behavior behind a flag that is only used by you.